pkg/api/validate/openshiftcluster_validatedynamic.go

```diff
@@ -319,30 +326,41 @@ func (dv *openShiftClusterDynamicValidator) validateSubnet(ctx context.Context,
 }
 
 // validateVnet checks that the vnet does not have custom dns servers set
+// and the subnets do not overlap with cluster pod/service CIDR blocks
 func (dv *openShiftClusterDynamicValidator) validateVnet(ctx context.Context, vnet *mgmtnetwork.VirtualNetwork) error {
 	dv.log.Print("validateVnet")
+	var err error
+	maxCIDRBlocks := len(dv.oc.Properties.WorkerProfiles) + 3
+	CIDRMap := make(map[string]*net.IPNet, maxCIDRBlocks)
```

Comment:
The reason I collect CIDR blocks into map is to make sure we look for overlap for unique subnets/blocks. For example when there are two workerProfiles on the same subnet, we do not want to put both of these into overlap function because the same subnet overlaps with itself.

Reply:
you can't create two subnets with the same case-insensitive name, so I don't think this is an issue here. I do like the idea of deduplicating them before getting here.

Reply:
You can't *create* like that but I think a user could subsequently add more machinesets which could cause the condition to happen.